Tests/RegexTests/ParseTests.swift

```diff
@@ -42,20 +42,33 @@ func parseTest(
   captures expectedCaptures: CaptureStructure = .empty,
   file: StaticString = #file,
   line: UInt = #line
+) {
+  parseTest(
+    input, .init(expectedAST, globalOptions: nil), syntax: syntax,
+    captures: expectedCaptures, file: file, line: line
+  )
+}
+
+func parseTest(
+  _ input: String, _ expectedAST: AST,
+  syntax: SyntaxOptions = .traditional,
+  captures expectedCaptures: CaptureStructure = .empty,
+  file: StaticString = #file,
+  line: UInt = #line
 ) {
   let ast = try! parse(input, syntax)
-  guard ast.root == expectedAST
-          || ast.root._dump() == expectedAST._dump() // EQ workaround
+  guard ast == expectedAST
+          || ast._dump() == expectedAST._dump() // EQ workaround
```

Comment:
Argh, this is still here! Any chance of adding a bool parameter for something like "ignore trivia", which we can approximate with the dump check? IIRC, this was mostly for trivia like non-semantic whitespace.

Reply:
Yeah this is awful, IIRC it's done to ignore both trivia and source location info. We could add a flag, though that wouldn't be sufficient to handle the character class trivia stripping which happens in `_dumpBase` in https://github.com/apple/swift-experimental-string-processing/pull/136. Really we ought to strip trivia from the entire tree prior to dumping (or if we strip enough, just compare with `==`), or otherwise have a parent type (like `PrettyPrinter`) that manages the dumping logic as it walks the tree and can store an `ignoreTrivia` flag. I can look into that in a follow-up.